### PR TITLE
tensile: init at 4.34.0-5.3.1

### PR DESCRIPTION
**Pull request by Madouura**
###### Description of changes
Tracking: #197885
Eighth part in a whole bunch of porting...
Mainly for incoming rocBLAS (have mercy on my PC...)

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by azahi** [comment]
Result of `nixpkgs-review pr 197719` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>tensile</li>
  </ul>
</details>

**Review by azahi — APPROVED** [review]

**Comment by Madouura** [comment]
After further investigation, particularly using ``rocBLAS``, I am fairly certain this can be merged as-is.
However, Tensile can use ``rocminfo``, ``rocm-smi``, and ``hip`` at runtime.
I am torn on whether I want to include these as ``propogatedBuildInputs``, so if anyone thinks I should or should not, please let me know.
In the meaning, marking this as ready on the tracker.

**Comment by Madouura** [comment]
IIRC from the source, llvm_rocm (clang and a couple other tools specifically) as well.

File: pkgs/development/libraries/tensile/default.nix [hunk 1]
```diff
@@ -0,0 +1,34 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, buildPythonPackage
+, pyyaml
+, msgpack
+, pandas
+, rocmVersion ? "5.3.1" # Tensile depends on specific ROCm versions per-release
```
Comment (lovesegfault) [inline pkgs/development/libraries/tensile/default.nix:8]: IMHO just use `rocm.version`, then users can just override by passing a newer rocm.

**Comment by lovesegfault** [comment]
Needs a rebase

**Comment by Madouura** [comment]
Rolled into #198770

**Pull request closed without merging**